### ISSUE TITLE
Stuttering View

### ISSUE THREAD

**Issue by prom6** [stijn-volckaert/udemo#135]
On demoplayback there is a minimal (horizontal) stutter. Barely noticeable when youre not looking for it, though I cant unsee it now. When actually playing the game its way smoother.
Maybe has something to do with the interpolation calculation?

Recorded this video on slomo and added a white bar for orientation, you can see on the floor texture (fullscreen) that it like moves some pixels left, some pixels right multiple times per second

https://github.com/stijn-volckaert/udemo/assets/71797130/c507c33a-2c56-4612-8c24-ed7ed4e40564

**Comment by SeriousBuggie** [comment]
When you move in game, you actually move, you stay on ground and for each frame can be calculated own value.
When you play demo you basically see as actor follow prdefined point with some interpolation between them. Actor fly on air, not walk on ground. And if you need make more data, for it be for each frame, best what you can use - previous data, until not come new from demo file.
So this not even interpolate from B to C, it is attempt predict where be C, before you get it, when you have A and B.
Demo not see what come next, so try predict it, and after it appear correct position. This make stuttering.
This goes by nature how thing goes in demo.
It act like client for real network server. So it not know what come next and when. Like it be with real net server.

This hard to fix. Demo play linear. For make proper interpolation between dots in time, need know when this appear. You need next point in time, but this point irregular and can appear in 1 seconds, in 1 hour or never at all.

**Comment by SeriousBuggie** [comment]
If you among v469d testers - please test build 247.
If no - you can join to them. Go to OldUnreal discord and say to Anthrax your github login.

**Comment by SeriousBuggie** [comment]
In latest test build add smooth for movement. By default it apply only for 3rdperson, since in first person view it make glitches, but you can try turn it on.

In general it caused by demo nature and how stored location of player. It stored frame by frame and use rounded locations. When playback, it just move you for this position, not prediction or interpolation here. So on low fps there can be visible micro movements. This can't be fixed.

Smooth too not able fix it good enough, since rotation still goes in quants and can't be smooth in good shape. However you can try experimental smooth and see what happen.

**Comment by prom6** [comment]
checked it in the latest release candidate. I feel its a bit better but still there. it wouldnt be as annoying if it wouldnt look like you dont move straight, it just looks like the player view is slightly rotated to the right and to compensate that the movement isnt straight but slightly forward-left and therefore the textures get oriented weirdly

**Comment by SeriousBuggie** [comment]
Unfortunately this can't be fixed. You can even input `slomo 0.1` for see it 10 times slower.
You can notice how player movement goes in quants. It update from time to time, when data come from demo file.

You can turn in udemo.ini smoothing for first person view, but result will be doubtful. Such can be fixed, during udemo nature.

Did you try with smooth for all variants?
`SmoothRecorderMovement=Smooth_All`

**Comment by prom6** [comment]
Yes I changed it to Smooth_All. Funnily there were no glitches when I set Smooth_All but when it only was set to 3rd Person i had some flickering going on.

Anyway, I appreciate your efforts and explaining why it happens, thanks for that

**Comment by SeriousBuggie** [comment]
It just workaround. It very depends how player in demo aim. If player in demo mostly aim forward and not do nice tracking on sides, Smooth_All look good.

But if it pro player which able track target on side, it look not really good. Better without. 

Such good visible in Pro BT demos. So by default it turned off.
